### PR TITLE
Support DocBase

### PR DESCRIPTION
**Pull request by swamp09**
Add support for [DocBase](https://docbase.io/) Webhook.

Link to details:  https://help.docbase.io/posts/78643

preview hooks:

### CODE REVIEW

File: spec/docbase_spec.rb [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    end
+
+    describe 'star_create event' do
+      let(:event_type) { 'star_create' }
+
+      it { expect(subject[:source]).to be_dom_equal <<~HTML }
+          <p>
+          <span><img src=\"https://secure.gravatar.com/avatar/9922bb39304a94c92e862211c23d60fc.png\" width=\"16\" height=\"16\" alt=\"\" /></span>
+          <a href='https://idobata.docbase.io/users/1'>eiwakun</a>
+          starrd <a href='https://idobata.docbase.io/posts/1'>hookの作り方</a>
```
Comment (hibariya) [inline spec/docbase_spec.rb]: seems misspelling for "starred" :star2:

Comment (swamp09) [reply spec/docbase_spec.rb]: 😇 
Thanks for review!
I fixed it and force pushed. ⭐️

File: lib/hooks/docbase/helper.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      def link_to_user(team_domain, user_id, user_name)
+        url = "https://#{team_domain}.docbase.io/users/#{user_id}"
+        capture_haml do
+          haml_tag :a, user_name, href: url
+        end
+      end
+
+      def link_to_joined_team_or_group(name, message)
+        url_first_index = message.index(/https:/)
+        url = message[url_first_index..-1]
```
Comment (ursm) [inline lib/hooks/docbase/helper.rb]: With `URI.extract` I think that the intention of the code will become clearer.

File: lib/hooks/docbase/hook.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      when 'post_publish'      then 'post_publish'
+      when 'post_update'       then 'post_update'
+      when 'post_group_update' then 'post_group_update'
+      when 'sharing_create'    then 'sharing_create'
+      when 'sharing_destroy'   then 'sharing_destroy'
+      when 'comment_create'    then 'comment_create'
+      when 'star_create'       then 'star_create'
+      when 'like_create'       then 'like_create'
+      when 'team_join'         then 'team_join'
+      when 'group_join'        then 'group_join'
```
Comment (ursm) [inline lib/hooks/docbase/hook.rb]: This case expression looks somewhat verbose. Should it be a bit more concise?

File: lib/hooks/docbase/helper.rb (hunk 2 above)
Comment (ursm) [reply lib/hooks/docbase/helper.rb]: Oops, `URI.extract` was deprecated. We could use `URI::RFC 2396_Parser#extract` instead.

File: lib/hooks/docbase/hook.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      else
+        'default'
```
Comment (ursm) [inline lib/hooks/docbase/hook.rb]: Whether or not to display unknown events needs to be taken into consideration, we basically ignore them in other hooks. Do you have any opinion?

File: lib/hooks/docbase/helper.rb (hunk 2 above)
Comment (swamp09) [reply lib/hooks/docbase/helper.rb]: Thank you for review! Fix to use URI::RFC2396_Parser#extract.

File: lib/hooks/docbase/hook.rb (hunk 3 above)
Comment (swamp09) [reply lib/hooks/docbase/hook.rb]: I fixed the verbose code.

File: lib/hooks/docbase/hook.rb (hunk 4 above)
Comment (swamp09) [reply lib/hooks/docbase/hook.rb]: I fixed it to ignore it in Docbase. 👍

**Comment by hibariya** [comment]
@swamp09 We've deployed this into production right now. Thank you for the great work :rocket:

**Pull request merged**